packages/block-editor/src/components/block-list/block-list-context.native.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * WordPress dependencies
+ */
+import { createContext, useContext } from '@wordpress/element';
+
+const blocksLayouts = { current: {} };
```

Comment:
> Not sure what's the purpose of defining the current property, I thought it was somehow related to a React reference, but I haven't found any reference to this with useRef or a ref prop. Hence, maybe we could drop the property and use blocksLayouts directly, wdyt?

Nice catch! I had this approach with the initial version and this is not needed anymore. I'll remove it 👍 

> On the other hand, I'm a bit concerned about having the blocks layout defined globally. Shouldn't this be defined when creating the BlockListProvider provider by the root block list?

I did this so `updateBlocksLayouts` would have direct access, but maybe we can pass the `blocksLayouts` reference as a param in `updateBlocksLayouts`, what do you think?

Reply:
> Nice catch! I had this approach with the initial version and this is not needed anymore. I'll remove it 👍

👍 

> I did this so updateBlocksLayouts would have direct access, but maybe we can pass the blocksLayouts reference as a param in updateBlocksLayouts, what do you think?

Yeah, I think we could do the same as for `findByRootId` and `findByRootId` where we pass the data (i.e. `blocksLayouts`). I'd like to note that having it globally is not bad per se from my POV, as I don't foresee that we'd have more than one instance of the blocks layouts. But since we're managing it by the context provider, maybe it makes more sense to let that be its source of truth.

Reply:
Awesome! Sounds good to me, I'll update it 😃

Reply:
I updated a bit around this including the suggestions from this [comment](https://github.com/WordPress/gutenberg/pull/39089#discussion_r818472988), but removing the `current` property causes issues when deleting a block's layout data, since its replacing the whole object [here](https://github.com/WordPress/gutenberg/pull/39089/files#diff-b6c828a514d1b7e0f2f05f83c96cd151b1b5ba7f93b854b73c74ddc1388d35caR85). So if we remove the `current` property and use the object directly it loses its reference.

Reply:
Ok, I see, in that case, it makes sense to keep the `current` property 👍 .